crates/cext/src/dag.rs

```diff
@@ -968,3 +971,96 @@ pub unsafe extern "C" fn qk_dag_topological_op_nodes(dag: *const DAGCircuit, out
         unsafe { out_order.add(i).write(node.index() as u32) }
     }
 }
+
+/// @ingroup QkDag
+/// Replace a node in a `QkDag` with a subcircuit specfied by another `QkDag`
+///
+/// @param dag A pointer to the DAG.
+/// @param node The node index of the operation to replace with the other `QkDag`. This
+///     must be the node index for an operation node in ``dag`` and the qargs and cargs
+///     count must match the number of qubits and clbits in `replacement`.
+/// @param replacement The other `QkDag` to replace `node` with. This dag must have
+///     the same number of qubits as the operation for ``node``. The node
+///     bit ordering will be ordering will be handled in order, so `qargs[0]` for
+///     `node` will be mapped to `qubits[0]` in `replacement`, `qargs[1]` to
+///     `qubits[0]`, etc. The same pattern applies to classical bits too.
```

Comment:
I didn't think about it in the meeting, but the reason that the map needs to exist is for nodes that have "extra" wires, like control-flow or classical expressions - those don't appear in `qargs` or `cargs`, so they don't have a native order that the user can be determined, and so they can't just build `replacement` in order.

If we merge this PR without it, we'll likely have to break the API later to add it back in.  That's not _necessarily_ a deal-breaker, but it is more than we thought of at the time.